include/seahorn/Expr/ExprOpNum.hh

```diff
@@ -22,18 +23,39 @@ enum class NumericOpKind {
   NINFTY,
   ITV
 };
+
+namespace typeCheck {
+namespace numType {
+
+  static inline Expr returnType (Expr exp, TypeChecker &tc) {
+      return tc.typeOf(exp->first());
+  }
+struct Unary {
+  static inline Expr inferType(Expr exp, TypeChecker &tc) {
+    return typeCheck::unary<INT_TY, REAL_TY, UNINT_TY>(exp, tc, returnType);
+  }
+};
+
+struct Nary {
+  static inline Expr inferType(Expr exp, TypeChecker &tc) {
+    return typeCheck::nary<INT_TY, REAL_TY, UNINT_TY>(exp, tc, returnType);
+  }
+};
+} // namespace numType
+} // namespace typeCheck
+
 // -- Numeric operators
 NOP_BASE(NumericOp)
 
-NOP(PLUS, "+", INFIX, NumericOp)
-NOP(MINUS, "-", INFIX, NumericOp)
-NOP(MULT, "*", INFIX, NumericOp)
-NOP(DIV, "/", INFIX, NumericOp)
-NOP(IDIV, "/", INFIX, NumericOp);
-NOP(MOD, "mod", INFIX, NumericOp)
-NOP(REM, "%", INFIX, NumericOp)
-NOP(UN_MINUS, "-", PREFIX, NumericOp)
-NOP(ABS, "abs", FUNCTIONAL, NumericOp)
+NOP_TYPECHECK(PLUS, "+", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(MINUS, "-", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(MULT, "*", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(DIV, "/", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(IDIV, "/", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(MOD, "mod", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(REM, "%", INFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(UN_MINUS, "-", PREFIX, NumericOp, typeCheck::numType::Nary)
+NOP_TYPECHECK(ABS, "abs", FUNCTIONAL, NumericOp, typeCheck::numType::Unary)
 
 NOP(PINFTY, "oo", PREFIX, NumericOp)
```

Comment:
Infinitys are used in intervals to mark lack of a bound, as in `[0, oo]` to represent all positive numbers. I don't think these expressions are used now. It is ok to leave them untyped (i.e., type as `any`)